### ISSUE TITLE
Out-of-bounds string read in SettingsDialog

### ISSUE THREAD

**Issue by jgcodes2020** [Rosalie241/RMG#298]
Trying to open settings (Ctrl+T) causes an error at line 362, apparently during a call to `QString::operator[]`.

https://github.com/Rosalie241/RMG/blob/69c24b9131909c338b03c48438744b253501146a/Source/RMG/UserInterface/Dialog/SettingsDialog.cpp#L355-L367

As far as I can tell, `pluginFileNames` is supposed to be an *array* of `QString`, so I have no idea what's going on.

**Comment by Rosalie241** [comment]
That's a nice catch, it's supposed to be
```cpp
if (pluginFileNames[index] == QString::fromStdString(p.File))
```
I copy pasted some code from the `commonPluginSettings` function, which did things slightly differently.

I'll try to push the fix as soon as I'm on my desktop again :heart:

**Comment by Rosalie241** [comment]
Fixed it with https://github.com/Rosalie241/RMG/commit/37839b0178b42b57a38a040bbf4e1f410e6952a1, I'll include it in the next release, thank you for reporting! :heart:

**Comment by Rosalie241** [comment]
Fixed in v0.6.7, thanks again :two_hearts:
